compiler/rustc_macros/src/symbols.rs

```diff
@@ -151,51 +149,99 @@ fn symbols_with_errors(input: TokenStream) -> (TokenStream, Vec<syn::Error>) {
         prev_key = Some((span, str.to_string()));
     };
 
+    let mut symbol_strings: Vec<String> = Vec::new();
+
     // Generate the listed keywords.
+    let mut keyword_stream = quote! {};
     for keyword in input.keywords.iter() {
         let name = &keyword.name;
         let value = &keyword.value;
         let value_string = value.value();
-        check_dup(keyword.name.span(), &value_string, &mut errors);
-        prefill_stream.extend(quote! {
-            #value,
-        });
+        let symbol_index = if check_dup(keyword.name.span(), &value_string, &mut errors).is_ok() {
+            let symbol_index = symbol_strings.len() as u32;
+            symbol_strings.push(value_string);
+            symbol_index
+        } else {
+            // Compilation is going to fail, because of a duplicate symbol name.
+            // Add a symbol definition, even one with a bogus value, to minimize
+            // unrelated errors.
+            0
```

Comment:
If any value works here, then `symbol_index` would work too, so it can be moved out of the `check_dup` condition.

Reply:
Done.